### PR TITLE
[KernelGen][Nvidia] add _standard_gamma_grad operator

### PR DESCRIPTION
**Pull request by NineAnnAnn**
## Summary
Adds a Triton kernel for `_standard_gamma_grad`. Computes gradients of the standard Gamma distribution with respect to the shape parameter.

## Testing
- Parametrized tests over alpha values, tensor shapes, and dtypes
- Validated against reference on device via `to_reference(inp, True)`
- 54 test cases covering various alpha values, edge cases, and gradient accuracy validation

## Performance
Test command: `pytest benchmark/test_standard_gamma_grad.py --level core` (NVIDIA H20)

### _standard_gamma_grad

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|--------------------|-------------------|---------|
| float16 | [1073741824], [1073741824] | 58.668 | 67.886 | 0.864 |
| float16 | [64, 64], [64, 64] | 0.019 | 0.096 | 0.195 |
| float16 | [4096, 4096], [4096, 4096] | 0.915 | 0.937 | 0.976 |
| float16 | [64, 512, 512], [64, 512, 512] | 0.918 | 0.937 | 0.979 |
| float16 | [1024, 1024, 1024], [1024, 1024, 1024] | 58.652 | 67.884 | 0.864 |
| float32 | [1073741824], [1073741824] | 57.977 | 67.121 | 0.864 |
| float32 | [64, 64], [64, 64] | 0.020 | 0.035 | 0.563 |
| float32 | [4096, 4096], [4096, 4096] | 0.905 | 0.937 | 0.967 |
| float32 | [64, 512, 512], [64, 512, 512] | 0.905 | 0.937 | 0.966 |
| float32 | [1024, 1024, 1024], [1024, 1024, 1024] | 58.017 | 67.174 | 0.864 |
| bfloat16 | [1073741824], [1073741824] | 57.964 | 68.382 | 0.848 |
| bfloat16 | [64, 64], [64, 64] | 0.019 | 0.034 | 0.538 |
| bfloat16 | [4096, 4096], [4096, 4096] | 0.907 | 0.933 | 0.972 |
| bfloat16 | [64, 512, 512], [64, 512, 512] | 0.906 | 0.933 | 0.971 |
| bfloat16 | [1024, 1024, 1024], [1024, 1024, 1024] | 57.970 | 68.424 | 0.847 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| _standard_gamma_grad | **0.819** |

## Files Changed
- `src/flag_gems/ops/_standard_gamma_grad.py`: Triton kernel implementation
- `tests/test_standard_gamma_grad.py`: Accuracy test
- `benchmark/test_standard_gamma_grad.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -13653,6 +13653,17 @@ ops:
...
+    description: Triton kernel implementation for _standard_gamma_grad.
+    for:
+      - _standard_gamma_grad
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Math
+    stages:
+      - alpha: '5.4'
```
Comment (Caeruleann) [inline conf/operators.yaml]: Please fix the formatting issue here.

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -244,6 +244,7 @@ def torch_ge(v):
     ("_softmax_backward_data", softmax_backward),
     ("_softmax_backward_data.out", softmax_backward_out),
     ("_sparse_semi_structured_mm", _sparse_semi_structured_mm),
+    ("_standard_gamma_grad", _standard_gamma_grad),
```
Comment (Caeruleann) [inline src/flag_gems/__init__.py]: There is no conflicting standard_gamma_grad ATen variant, so the FlagGems API should drop the leading underscore.

File: src/flag_gems/ops/_standard_gamma_grad.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    result = tl.where(x < 0.8, small, tl.where(alpha > 8.0, large, rational))
+    return result
+
+
+@pointwise_dynamic(promotion_methods=[(0, 1, "DEFAULT")])
+@triton.jit
+def _standard_gamma_grad_func(self_grad, output):
+    alpha = self_grad.to(tl.float32)
+    x = output.to(tl.float32)
+    return _standard_gamma_grad_one(alpha, x).to(self_grad.dtype)
```
Comment (Caeruleann) [inline src/flag_gems/ops/_standard_gamma_grad.py:143]: Both inputs are unconditionally downcast to FP32, so FP64 inputs only receive FP32 computation accuracy. Please preserve double-precision computation for FP64 inputs, or explicitly reject FP64, and add an ALL_FLOAT_DTYPES test covering the branch boundaries.

File: src/flag_gems/ops/_standard_gamma_grad.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    p = cv0 + v * (cv1 + v * (cv2 + v * cv3))
+    q = cv4 + v * (cv5 + v * (cv6 + v * cv7))
+    rational = tl.exp(p / q)
+
+    # Select the branch matching ATen's control flow.
+    result = tl.where(x < 0.8, small, tl.where(alpha > 8.0, large, rational))
+    return result
+
+
+@pointwise_dynamic(promotion_methods=[(0, 1, "DEFAULT")])
```
Comment (Caeruleann) [inline src/flag_gems/ops/_standard_gamma_grad.py]: DEFAULT promotion changes the ATen dtype contract: mixed dtypes are promoted and integer inputs may reach the kernel, while native _standard_gamma_grad requires matching floating dtypes. Please reject unsupported or mismatched dtypes and add error-parity tests.

File: tests/test_standard_gamma_grad.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    ref_out = torch._standard_gamma_grad(ref_self_grad, ref_output)
+    res_out = flag_gems._standard_gamma_grad(res_self_grad, res_output)
+
+    utils.gems_assert_close(res_out, ref_out, dtype)
+
+
+@pytest.mark.standard_gamma_grad
+@pytest.mark.parametrize("shape", utils.POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_standard_gamma_grad_near_mode(shape, dtype):
```
Comment (Caeruleann) [inline tests/test_standard_gamma_grad.py:30]: Please also cover the alpha > 8 away-from-mode branch with values below 0.9 * alpha and above 1.1 * alpha, as the current large-alpha test only exercises the near-mode path.

**Comment by NineAnnAnn** [comment]
Thanks for the review. I've addressed all five comments and rebased onto the latest `master` (no conflicts):

1. **`conf/operators.yaml` formatting** — the entry is now a top-level sibling `- id: standard_gamma_grad` (previously nested under the preceding op's `stages:`).

2. **Drop leading underscore for the FlagGems API** — the function is now `standard_gamma_grad` (no conflicting non-underscore ATen variant). The `_FULL_CONFIG` registration key stays the aten name `_standard_gamma_grad`, and the public op is exported/called as `flag_gems.standard_gamma_grad`.

3. **Preserve fp64 precision** — the kernel now branches on `self_grad.dtype.is_fp64()`: fp64 inputs compute in `tl.float64`, while fp16/bf16/fp32 compute in `tl.float32` (mirroring ATen's per-dtype computation). Tests now use `ALL_FLOAT_DTYPES` so the fp64 path is covered. (Also fixed an fp64 `libdevice pow` dispatch by typing the `-1.5` exponent to the compute dtype.)

4. **Dtype contract / reject unsupported dtypes** — the wrapper now rejects non-floating inputs and mismatched dtypes with `RuntimeError`, matching native `_standard_gamma_grad` instead of relying on `DEFAULT` promotion. Added error-parity tests for mixed-dtype and integer inputs (asserting both torch and FlagGems raise).

5. **alpha > 8 away-from-mode coverage** — added `test_standard_gamma_grad_away_from_mode`, which builds outputs strictly below `0.9 * alpha` and above `1.1 * alpha` so the away-from-mode sub-case of the Rice saddle-point branch is exercised (the existing large-alpha test only hit the near-mode path).

Local `tests/test_standard_gamma_grad.py` passes (98/98) and all pre-commit hooks pass.

**Review by Caeruleann — APPROVED** [review]
